extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -69,16 +72,20 @@ export class AmpShareTracking extends AMP.BaseElement {
     this.vendorHref_ = this.element.getAttribute('data-href');
     dev().fine(TAG, 'vendorHref_: ', this.vendorHref_);
 
-    this.shareTrackingFragments_ = Promise.all([
-      this.getIncomingFragment_(),
-      this.getOutgoingFragment_()]).then(results => {
-        dev().fine(TAG, 'incomingFragment: ', results[0]);
-        dev().fine(TAG, 'outgoingFragment: ', results[1]);
-        return {
-          incomingFragment: results[0],
-          outgoingFragment: results[1],
-        };
-      });
+    this.shareTrackingFragments_ = Promise.all(
+      [this.getIncomingFragment_(), this.getOutgoingFragment_()]
+    ).then(results => {
+      const incomingFragment = results[0];
+      const outgoingFragment = results[1];
+      dev().fine(TAG, 'incomingFragment: ', incomingFragment);
+      dev().fine(TAG, 'outgoingFragment: ', outgoingFragment);
+      if (outgoingFragment) {
+        const newFragment = this.getNewViewerFragment_(incomingFragment,
+            outgoingFragment);
+        viewerForDoc(this.getAmpDoc()).updateFragment(newFragment);
```

Comment:
Let's make sure `newFragment` has a leading `#`, since we expect the fragment request to have it.